pandas/core/util/hashing.py

```diff
@@ -166,33 +194,33 @@ def hash_tuples(vals, encoding="utf8", hash_key: str = _default_hash_key):
     )
 
     if not isinstance(vals, ABCMultiIndex):
-        vals = MultiIndex.from_tuples(vals)
+        mi = MultiIndex.from_tuples(vals)
```

Comment:
side note... `tuples` parameter of MultiIndex.from_tuples is untyped. I think that sometimes it is better to keep typing PRs small but ensure all lower level functions are typed first

Reply:
ive got a branch going that is doing MultiIndex annotations, will update there